internal/driver/fetch.go

```diff
@@ -356,6 +358,13 @@ func fetch(source string, duration, timeout time.Duration, ui plugin.UI) (p *pro
 		}
 		f, err = fetchURL(sourceURL, timeout)
 		src = sourceURL
+	} else if isPerf, isPerfErr := isPerfFile(source); isPerf {
+		// Since the if statement is a new scope, if isPerfErr is named
```

Comment:
If you have isPerfFile only return a bool (==false in case of errors) you'll save a lot of this complexity.